helix-core/src/test/java/org/apache/helix/TestHelper.java

```diff
@@ -278,16 +281,18 @@ public static void setupCluster(String clusterName, String zkAddr, int startPort
   public static void setupCluster(String clusterName, String zkAddr, int startPort,
       String participantNamePrefix, String resourceNamePrefix, int resourceNb, int partitionNb,
       int nodesNb, int replica, String stateModelDef, RebalanceMode mode, boolean doRebalance) {
-    HelixZkClient zkClient = SharedZkClientFactory.getInstance()
-        .buildZkClient(new HelixZkClient.ZkConnectionConfig(zkAddr));
+    HelixZkClient zkClient = createZkClient(zkAddr);
     try {
       if (zkClient.exists("/" + clusterName)) {
         LOG.warn("Cluster already exists:" + clusterName + ". Deleting it");
         zkClient.deleteRecursively("/" + clusterName);
       }
 
       ClusterSetup setupTool = new ClusterSetup(zkAddr);
-      setupTool.addCluster(clusterName, true);
+      List<BuiltInStateModelDefinitions> stateModelDefinitions = Objects.isNull(stateModelDef)
+          ? (resourceNb == 0 ? Collections.emptyList() : Arrays.asList(BuiltInStateModelDefinitions.values()))
+          : Arrays.asList(BuiltInStateModelDefinitions.valueOf(stateModelDef));
+      setupTool.addCluster(clusterName, stateModelDefinitions, false);
```

Comment:
Same here. I don't know why we would like to have tests on non existing state model?

Reply:
Yes, updated the reasoning with a comment as well along here. This is needed to avoid loading all the state-model defs and loading only the one which user has explicitly specified. Otherwise, we will load all.